2009-02-13  Michael Tautschnig  <mt@debian.org>

	* bin/faimond: Try to load Proc::Daemon only in case -d is given; if it fails,
		print a helpful error message (closes: #513090)
Index: trunk/bin/faimond
===================================================================
--- trunk.orig/bin/faimond
+++ trunk/bin/faimond	
@@ -15,7 +15,6 @@
 use strict;
 use Socket;
 use Getopt::Std;
-use Proc::Daemon;
 
 $| = 1;
 my ($port, $timeout, $daemon, $timestamp);
@@ -78,7 +77,7 @@
       }
       close(PIDFILE);
     }
-    Proc::Daemon::Init;
+    eval "Proc::Daemon::Init";
     umask 022;
 
     open(PIDFILE, '>', "$pidfile") or die "open $pidfile: $!";
@@ -177,6 +176,8 @@
 }
 
 if (defined($opt_d)) {
+  (eval "require Proc::Daemon") or
+    die "Daemon mode not available, Proc::Daemon not found. Please install libproc-daemon-perl\n";
   # If in daemon mode, use standard daemon log file
   $logfile = $daemonlogfile;
 }
